PROJECT

```diff
@@ -0,0 +1,20 @@
+# Code generated by tool. DO NOT EDIT.
+# This file is used to track the info used to scaffold your project
+# and allow the plugins properly work.
+# More info: https://book.kubebuilder.io/reference/project-config.html
+domain: kubernetes.azure.com
+layout:
+- go.kubebuilder.io/v4
+projectName: app-routing
+repo: github.com/Azure/aks-app-routing-operator
+resources:
+- api:
+    crdVersion: v1
+    namespaced: false
+  controller: true
+  domain: kubernetes.azure.com
```

Comment:
It's used by [aks labels](https://learn.microsoft.com/en-us/azure/aks/use-labels#reserved-system-labels). 

We use it for our annotations in [app routing](https://learn.microsoft.com/en-us/azure/aks/app-routing?tabs=with-osm)
>  kubernetes.azure.com/tls-cert-keyvault-uri: <KeyVaultCertificateUri>
    kubernetes.azure.com/use-osm-mtls: "true"

Our crd domain will be `approuting.kubernetes.azure.com` because it's the group + domain.

Other AKS Operators follow similar patterns.

ASO uses the domain `resources.azure.com`. This is a bit different from us because ASO as a product isn't supposed to only run on AKS (it can run on other kubernetes infrastructure). Also, this fits them better because they have a CRD for each kind of azure resource like `batchaccounts.batch.azure.com`. Batch accounts aren't related to kubernetes so it would be weird for them to have kubernetes in the domain.

Fleet uses `fleet.azure.com`. Again, it's a little different because Fleet is a top-level RP in azure. Fleet is at a higher level than k8s.

Using the `approuting.kubernetes.azure.com` conveys that app routing is a product inside aks inside azure which fits us and fits our annotations.